src/commands/ingressCommands.ts

```diff
@@ -61,15 +61,21 @@ export async function toggleIngress(context: IActionContext, node?: IngressTreeI
     await updateIngressSettings(context, { ingress, node, working, workCompleted });
 }
 
-export async function editTargetPort(context: IActionContext, node?: IngressTreeItem): Promise<void> {
-    if (!node) {
-        node = await ext.tree.showTreeItemPicker<IngressTreeItem>(IngressTreeItem.contextValue, context);
+export async function editTargetPort(context: IActionContext, target?: IngressTreeItem | GenericTreeItem): Promise<void> {
+    if (!target) {
+        target = await ext.tree.showTreeItemPicker<IngressTreeItem>(IngressTreeItem.contextValue, context);
     }
 
+    // GenericTreeItem will be a targetPort node
+    const node: IngressTreeItem = target instanceof IngressTreeItem ? target : target.parent as IngressTreeItem;
+
     const title: string = localize('updateTargetPort', 'Update Target Port');
     const promptSteps: AzureWizardPromptStep<IContainerAppContext>[] = [new TargetPortStep()];
 
-    const wizardContext: IContainerAppContext = { ...context, ...node.subscription, managedEnvironmentId: node.parent.managedEnvironmentId };
+    const wizardContext: IContainerAppContext = {
+        ...context, ...node.subscription,
+        managedEnvironmentId: node.parent.managedEnvironmentId, defaultPort: node.data.targetPort
```

Comment:
I'd split these into two lines, at a glance it's easy to miss the comma. Maybe also split the above `...context, ...node.subscription,` just for consistency.